.github/actions/cache-save/action.yml

```diff
@@ -0,0 +1,51 @@
+name: "Save download caches conditionally"
+description: "Conditionally saves external dependencies caches based on cache-hit status."
+inputs:
+  repo-cache-hit:
+    description: 'Cache hit status for Bazel Repo Cache'
+    required: true
+  repo-cache-dir:
+    description: 'The Bazel Repo cache directory path'
+    required: true
+    default: '~/repo-cache'
+  go-mod-cache-hit:
+    description: 'Cache hit status for Go Mod Cache'
+    required: true
+  go-mod-cache-dir:
+    description: 'The Go Mod cache directory path'
+    required: true
+    default: '/home/runner/go-mod-cache'
+  yarn-cache-hit:
+    description: 'Cache hit status for Yarn Cache'
+    required: true
+  yarn-cache-dir:
+    description: 'The Yarn cache directory path'
+    required: true
+    default: ~/.cache/yarn/v6/
+runs:
+  using: "composite"
+  steps:
+    - name: Save Bazel Repo Cache
+      uses: actions/cache/save@v4
+      # Some workflows may contribute more to the cache
+      # if: ${{ inputs.repo-cache-hit != 'true' }}
```

Comment:
nit: Remove?

Reply:
I replaced it with a comment explaining why we are using `always()` here.